lib/network_monitor.c

```diff
@@ -1015,14 +1015,36 @@ static void check_default_route(struct nm_addr_info *ai)
          */
         mptcpd_addr_get(ai);
 
-        if (l_netlink_send(ai->nm->rtnl,
-                           RTM_GETROUTE,
-                           0,
-                           &store,
-                           buf - (char *) &store,
-                           handle_rtm_getroute,
-                           ai,
-                           NULL) == 0) {
+	unsigned int result = 0;
+
+#ifdef HAVE_L_NETLINK_MESSAGE_NEW_SIZED
+	// ELL >= 0.68
+        ptrdiff_t const msg_size = buf - (char *) &store;
+        struct l_netlink_message *const nlm =
+                l_netlink_message_new_sized(RTM_GETROUTE,
+                                            0,
+                                            msg_size);
+
+        l_netlink_message_add_header(nlm, &store, msg_size);
+
+	result = l_netlink_send(ai->nm->rtnl,
+				nlm,
+				handle_rtm_getroute,
+				ai,
+				NULL);
+#else
+	// ELL < 0.68
+	result = l_netlink_send(ai->nm->rtnl,
+				RTM_GETROUTE,
+				0,
+				&store,
+				buf - (char *) &store,
+				handle_rtm_getroute,
+				ai,
+				NULL);
+#endif
```

Comment:
> I'm not happy with the `#ifdef` blocks introduced by this change, but it should at least get mptcpd working with ELL 0.68.

@ossama-othman : would you be happier to move this code to a dedicated helper taking the same argument as `l_netlink_send` before v0.68, and handling the two versions? The modifications in the code here would be minimal, only renaming `l_netlink_send` to the new helper name.

Reply:
Done!  Thanks for the excellent suggestion @matttbe!